### PR TITLE
Bug 1194092 - [browser] Firefox Account sync history

### PR DESCRIPTION
**Pull request by danhuang1202**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c7f8916fd98193b57d13090741db4d6c141a217a

File: tv_apps/browser/js/awesomescreen.js [hunk 1]
```diff
@@ -1334,16 +1119,23 @@ var Awesomescreen = {
...
+      if(this.isDisplayedTop()){
         type = this.DEFAULT_TOPSITE;
       }
     }
 
     switch(type){
+//IFDEF_FIREFOX_SYNC
+      case this.FIREFOX_SYNC_BOOKMARK:
+      case this.FIREFOX_SYNC_HISTORY:
+//ENDIF_FIREFOX_SYNC
```
Comment (begeeben) [inline tv_apps/browser/js/awesomescreen.js]: The ```//ENDIF_FIREFOX_SYNC``` should be move to line 1138.

File: tv_apps/browser/js/awesomescreen.js [hunk 2]
```diff
@@ -2009,11 +1719,10 @@ var Awesomescreen = {
...
         this.callAwesomescreenEvents(url);
         break;
-      case this.isDisplayedList() :
+      case this.historyList.isDisplay():
         this.showAwesomeLoadingIcon();
-        url = this.selectList.childNodes[1].childNodes[1].textContent;
-        this.pintohomeTitle =
-          this.selectList.childNodes[1].childNodes[0].textContent;
+        url = this.historyList.getFocusItemUri();
+        this.pintohomeTitle = this.bookmarkList.getFocusItemTitle();
```
Comment (begeeben) [inline tv_apps/browser/js/awesomescreen.js]: ```this.historyList.getFocusItemTitle();```

File: tv_apps/browser/js/historyStore.js [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+
+    isMoreData: false,
+
+    cache: new Map(),
+
+    cacheIndexStartAt: 0,
+
+    cacheIndexEndAt: 0,
+
+    lastDataIndex: undefined,
```
Comment (begeeben) [inline tv_apps/browser/js/historyStore.js]: Please change it to ```null```.

File: tv_apps/browser/js/historyStore.js [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+
+    currentFolder : undefined,
```
Comment (begeeben) [inline tv_apps/browser/js/historyStore.js]: Same as above.

File: tv_apps/browser/js/historyStore.js [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+
+    reset: function (cb) {
+      return new Promise(resolve => {
+        this.isMoreDat = false;
```
Comment (begeeben) [inline tv_apps/browser/js/historyStore.js]: Typo ```isMoreData```.

File: tv_apps/browser/js/historyStore.js [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+        this.cache.clear();
+        this.cacheIndexStartAt = 0;
+        this.cacheIndexEndAt = 0;
+        this.lastDataIndex = undefined;
+        this.currentFolder = undefined;
```
Comment (begeeben) [inline tv_apps/browser/js/historyStore.js]: Please change these to ```null```.

File: tv_apps/browser/js/historyStore.js (hunk 5 above)
Comment (begeeben) [inline tv_apps/browser/js/historyStore.js]: Please add ```//IFNDEF_FIREFOX_SYNC``` like ```bookmarkStore.js```.

File: tv_apps/browser/js/historyStore.js [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+        this.currentFolder = folderId;
+        this.fetchCache().then(() => {
+          cb(getCaches());
+        });
+      } else {
+        cb(getCaches());
+      }
+    },
+
+    getByIndex: function (index, folderId, cb) {
```
Comment (begeeben) [inline tv_apps/browser/js/historyStore.js]: Please discuss ```getByIndex``` and ```fetchCache``` with me.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=71da694670cfe23c822d7d2a211d26166b5c8be8

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2ed3cf589c45c490bffac80dd967febcfdb34cf4

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9c5d769fe729875f55fbd8ba5794fe35c2278f8a

**Pull request merged**